pkg/provider/chef/chef.go

```diff
@@ -59,22 +58,11 @@ func init() {
 
 func (providerchef *Providerchef) NewClient(ctx context.Context, store v1beta1.GenericStore, kube kclient.Client, namespace string) (v1beta1.SecretsClient, error) {
 	storeSpec := store.GetSpec()
```

Comment:
lets move this `storeSpec := store.GetSpec()`  inside `func commonValidation(store v1beta1.GenericStore)` and return `storeSpec *v1beta1.SecretStoreSpec` type from `commonValidation`  function after validating

Reply:
I understand your point,  that way the storeSpec:= store.GetSpec() also will be done at one single place rather than twice in two different functions.
But wouldn’t it be better if a validation function returns only an error.

Reply:
from `store` , which is of type v1beta1.GenericStore only we extract all other values ..
so would better to validate like func commonValidation(store v1beta1.GenericStore) and return error ..

Reply:
Changed the name of the commonValidation function to getChefProvider which now runs the validation on the provided SecretStore and also returns the ChefProvider from the store.